cardano-crypto-tests/src/Test/Crypto/EllipticCurve.hs

```diff
@@ -100,6 +102,19 @@ testBLSCurve name _ =
         BLS.blsIsInf (BLS.blsMult a BLS.scalarPeriod)
     , testProperty "mult by p+1 is identity" $ \(a :: BLS.Point curve) ->
         BLS.blsMult a (BLS.scalarPeriod + 1) === a
+    , testProperty "scalar mult associative" $ \(a :: BLS.Point curve) (b :: Integer) (c :: Integer) ->
```

Comment:
I've added it in d54b6cb. Could you check that it is what you expected? I had to add `
{-# OPTIONS_GHC -fno-warn-type-defaults #-}` ghc flag for (AFAIU) defaulting the integral type of `choose`.

Reply:
Disabling warnings should be done as a last resort. The reason why it is complaining is not because of `choose`, but because of `^`, which is polymorphic in both arguments. What you need is `2 ^ (128 :: Int)`